### PR TITLE
Make sure the TIP's tag is updated in the storage too

### LINKED ISSUE #783: Chain head storage tag not kept up to date
Currently, there is a tag in storage for the main branch, but it is only ever set to the block0's hash. 

The tag: 
https://github.com/input-output-hk/jormungandr/blob/646b07116e2f40be66aee51df5e353c56b4fdd9e/jormungandr/src/blockchain/chain.rs#L106

Expected behavior: The tag to be kept up to date when the chain tip changes.

comments from discussion with @NicolasDP about it:

> Every time the tip is updated it should update the storage tag for this too.
> It’s only done for the block0 hash.

### LINKED ISSUE #788: not yet implemented: method to load a Ref from the storage is not yet there
**Describe the bug**
When restarting a node with trusted peer this happen:

```
Sep 17 21:20:44.134 INFO Starting jormungandr 0.3.3 (, release, linux [x86_64]) - [rustc 1.36.0 (a53f9df32 2019-07-03)], task: init
Sep 17 21:20:44.134 INFO storing blockchain in '"./storage/blocks.sqlite"', task: init
Sep 17 21:20:44.151 INFO connecting to bootstrap peer 3.122.73.200:3000, peer: /ip4/3.122.73.200/tcp/3000, network: bootstrap, task: bootstrap
thread 'main' panicked at 'not yet implemented: method to load a Ref from the storage is not yet there', jormungandr/src/blockchain/chain.rs:259:37
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
   1: std::sys_common::backtrace::_print
   2: std::panicking::default_hook::{{closure}}
   3: std::panicking::default_hook
   4: std::panicking::rust_panic_with_hook
   5: std::panicking::continue_panic_fmt
   6: std::panicking::begin_panic_fmt
   7: futures::future::chain::Chain<A,B,C>::poll
   8: futures::future::chain::Chain<A,B,C>::poll
   9: futures::future::chain::Chain<A,B,C>::poll
  10: futures::future::chain::Chain<A,B,C>::poll
  11: futures::future::chain::Chain<A,B,C>::poll
  12: futures::future::chain::Chain<A,B,C>::poll
  13: futures::future::chain::Chain<A,B,C>::poll
  14: <futures::stream::fold::Fold<S,F,Fut,T> as futures::future::Future>::poll
  15: futures::future::chain::Chain<A,B,C>::poll
  16: futures::future::chain::Chain<A,B,C>::poll
  17: futures::future::chain::Chain<A,B,C>::poll
  18: futures::task_impl::std::set
  19: std::thread::local::LocalKey<T>::with
  20: tokio_current_thread::Entered<P>::block_on
  21: std::thread::local::LocalKey<T>::with
  22: std::thread::local::LocalKey<T>::with
  23: std::thread::local::LocalKey<T>::with
  24: std::thread::local::LocalKey<T>::with
  25: tokio::runtime::current_thread::runtime::Runtime::block_on
  26: tokio::runtime::current_thread::block_on_all
  27: jormungandr::network::bootstrap::bootstrap_from_peer
  28: jormungandr::network::bootstrap
  29: jormungandr::main
  30: std::rt::lang_start::{{closure}}
  31: std::panicking::try::do_call
  32: __rust_maybe_catch_panic
  33: std::rt::lang_start_internal
  34: main
  35: __libc_start_main
  36: _start
```

**Mandatory Information**
running revision 646b07116e2f40be66aee51df5e353c56b4fdd9e.

**To Reproduce**
1.
`nix-shell https://github.com/input-output-hk/jormungandr-nix/archive/listen-address.tar.gz --argstr trusted_peers /ip4/3.122.73.200/tcp/3000`
2.
`mkdir storage`
3.
uncompress/copy attached sqlite db into the storage folder
4.
`RUST_BACKTRACE=1 run-jormungandr`

**Expected behavior**
the node start correctly.

**Additional context**
the genesis block of the cluster was generated using the following command:
`nix-shell https://github.com/input-output-hk/jormungandr-nix/archive/listen-address.tar.gz --arg numberOfStakePools 7 --arg faucetAmounts '[ 1000000000 1000000000 1000000000 1000000000 1000000000 1000000000 1000000000 100000000 100000000 100000000 ]' --arg slots_per_epoch 21600 --arg slot_duration 20 --arg consensus_genesis_praos_active_slot_coeff 0.1 --arg kes_update_speed 86400 --arg numberOfLeaders 7`

[blocks.zip](https://github.com/input-output-hk/jormungandr/files/3623115/blocks.zip)

### PR DESCRIPTION
**Pull request by NicolasDP**
Everytime the TIP is updated we also need to update the storage's tag for the main branch.

fixes #783 
fixes #788

### CODE REVIEW

**Comment by mzabaluev** [comment]
The tests hit #788, it seems.

**Comment by NicolasDP** [comment]
no, not the same. Here the error is when loading from the storage

File: jormungandr/src/network/bootstrap.rs [hunk 1]
```diff
@@ -83,7 +92,18 @@ pub fn bootstrap_from_peer(
...
-        .and_then(move |tip| branch.update_ref(tip).map_err(|_| unreachable!()));
+        .and_then(move |tip| {
+            branch
+                .update_ref(Arc::clone(&tip))
+                .map_err(|_| unreachable!())
+                .map(|_prev_tip| tip)
+        })
+        .and_then(move |tip| {
+            storage
+                .put_tag(MAIN_BRANCH_TAG.to_owned(), tip.hash())
```
Comment (mzabaluev) [inline jormungandr/src/network/bootstrap.rs]: `put_tag` could be extended to accept an `Into<String>` as a tag name.

**Comment by NicolasDP** [comment]
The test didn't hit #788 but it was similar. The actual issue was a misuse of `get_ref` in the case we were loading blocks in memory that we already have in storage.

So the actual problem was when we were bootstrapping (from the network or from the storage). Because we are validating the blocks before loading them in memory we were seeing the issue of utilising `get_ref` to check if the block was already in storage or not. In this case we only want to check if the block is yet again valid.

File: jormungandr/src/network/bootstrap.rs (hunk 1 above)
Comment (NicolasDP) [reply jormungandr/src/network/bootstrap.rs]: I've created an issue to update the API #791

**Pull request merged**